csrc/utils.h

```diff
@@ -621,86 +623,205 @@ void checkAllEqual(std::initializer_list<T> elements) {
 
 #if __cplusplus >= 202302L
 
+using std::views::enumerate;
 using std::views::zip;
 
 #else
 
 namespace views {
-
-template <std::ranges::input_range... Rs>
+template <std::ranges::view... Rs>
 class zip_view : public std::ranges::view_interface<zip_view<Rs...>> {
  private:
   std::tuple<Rs...> bases;
 
-  // Iterator for begin()
-  struct begin_iterator {
+  static constexpr bool is_bidirectional =
+      (std::ranges::bidirectional_range<Rs> && ...);
```

Comment:
Got it -- all ranges have to be bidirectional for the zip_view to be

Reply:
Yes, exactly